### PR TITLE
Add example rule

### PR DESCRIPTION
**Pull request by slavcho-slavchev-hx**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
+// TODO: add rule for `.restore` being called on methods, suggest it gets called on sinon or sandbox only
+
 module.exports = {
   rules: {
+    'stubbing-in-nested-before-each': require('./rules/stubbingInNestedBeforeEach'),
```
Comment (hx-emanolov) [inline index.js]: Maybe `no-stubbing-in-nested-before-each`?
https://eslint.org/docs/developer-guide/working-with-rules#rule-naming-conventions

File: index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   rules: {
+    'no-stubbing-in-nested-before-each': require('./rules/noStubbingInNestedBeforeEach'),
```
Comment (hx-emanolov) [inline index.js]: Maybe `noStubbingInNestedBeforeEach.js` -> `no-stubbing-in-nested-before-each.js`?
https://github.com/eslint/eslint/tree/master/lib/rules

Comment (slavcho-slavchev-hx) [reply index.js]: I was actually thinking of that, great shout tho 👍

File: README.md [hunk 3]
```diff
@@ -28,3 +28,72 @@ npm link ../eslint-plugin-hx-rules
     }
 }
 ```
+
+### Rules
+
+* [No stubbing in nested before each](#no-stubbing-in-nested-before-each)
+
+##### No stubbing in nested before each
```
Comment (hx-emanolov) [inline README.md]: How about moving this doc into `./docs/rules/no-stubbing-in-nested-before-each.md` and linking to that file here instead?

Example:
https://github.com/eslint/eslint/tree/master/docs/rules

Comment (slavcho-slavchev-hx) [reply README.md]: Could probably even have a separate doc file for each rule, alongside the js file (index.js) file.

**Review by hx-emanolov — APPROVED** [review]
:shipit:

**Pull request merged**